### PR TITLE
Full support for all node-sass options

### PR DESCRIPTION
**Pull request by teodragovic**
Hi,

I made minor tweak to code so all user options are passed to `render()` method. This allows to use all node-sass options like setting precision factor, source comments, embeded source maps, add custom functions etc.

### CODE REVIEW

File: middleware.js [hunk 1]
```diff
@@ -83,10 +84,7 @@ module.exports = function(options) {
   // Enable debug output
   var debug = options.debug;
 
-  var indentedSyntax = options.indentedSyntax || null;
-  var sassExtension = (indentedSyntax === true) ? '.sass' : '.scss';
-
-  var sourceMap = options.sourceMap || null;
```
Comment (am11) [inline middleware.js]: Revert change on line 89, it is used on [line 175](https://github.com/teodragovic/node-sass-middleware/blob/master/middleware.js#L175).

**Comment by am11** [comment]
Thanks @teodragovic!
I have left a comment, as the CI build is failing.
Otherwise LGTM.

@kevva, @iMoritz, @Xiphe, what do you guys think about this idea?

**Comment by Xiphe** [comment]
no time to check right now, sorry

**Comment by teodragovic** [comment]
@am11 I reverted `sourceMap` line but build is still failing with `Uncaught SyntaxError: Unexpected token E` message when output has errors. Could you take a look? Sorry I didn't run tests before.

**Comment by am11** [comment]
@teodragovic, thanks for fixing the `sourceMap` issue. 

This delta will fix the build:

```diff
diff --git a/middleware.js b/middleware.js
index 981e127..e33bd60 100644
--- a/middleware.js
+++ b/middleware.js
@@ -62,9 +62,10 @@ module.exports = function(options) {
   var sassMiddlewareError = null;

   // This function will be called if something goes wrong
+  var cachedErrorCb = options.error;
   var error = function(err) {
-    if (options.error) {
-      options.error(err);
+    if (cachedErrorCb) {
+      cachedErrorCb(err);
     }

     sassMiddlewareError = err;
```

The reason being; `options.error` and `options.success` are two conflicting  members of options which downstreams should cache before forwarding the object to node-sass. Since JS objects are mutable, those get overwritten. There is history attached to those two members; we used to have those (jQuery) style callbacks, now we use node.js like callback. see https://github.com/sass/node-sass/issues/676. 

The libsass [binding call-point](https://github.com/sass/node-sass/blob/34db870992bcae17de30dd0e1f0fd8ddb4bcc363/lib/index.js#L275) hasn't been updated. (although not super-important but) IMO, it can be modified to avoid the internal dependency on `option.*`.

FYI @xzyfer, @saper

**Comment by iMoritz** [comment]
Except for the bug LGTM too :+1:

**Comment by teodragovic** [comment]
@am11 thanks for clarifying the issue. I made corrections, now everything seems to be working correctly.

**Comment by am11** [comment]
Thanks a bunch! :tada:

**Pull request merged**